opencti-platform/opencti-front/src/private/components/analyses/external_references/ExternalReferencePopoverDeletion.tsx

```diff
@@ -0,0 +1,141 @@
+    values: { entity_type: t_i18n('entity_External-Reference') },
+  });
+  const [commit] = useApiMutation(
+    externalReferencePopoverDeletionDeleteMutation,
+    undefined,
+    { successMessage: deleteSuccessMessage },
+  );
+  const handleOpenDelete = () => {
+    setDisplayDelete(true);
+  };
+  const handleCloseDelete = () => {
+    setDisplayDelete(false);
+  };
+  const submitDelete = () => {
+    setDeleting(true);
+    commit({
+      variables: {
+        id,
+      },
+      updater: (store) => {
+        if (handleRemove && objectId) {
+          deleteNodeFromId(
+            store,
+            objectId,
+            'Pagination_externalReferences',
+            undefined,
+            id,
+          );
+        }
+      },
+      onCompleted: () => {
+        setDeleting(false);
+        if (handleRemove) {
+          handleCloseDelete();
+        } else {
+          navigate('/dashboard/analyses/external_references');
+        }
+      },
+    });
+  };
+  return (
+    <React.Fragment>
+      <Security needs={[KNOWLEDGE_KNUPDATE_KNDELETE]}>
+        <Button
+          color="error"
+          variant="contained"
+          onClick={handleOpenDelete}
+          disabled={deleting}
+          sx={{ marginTop: 2 }}
+        >
+          {t_i18n('Delete')}
+        </Button>
+      </Security>
+      <Dialog
+        PaperProps={{ elevation: 1 }}
+        open={displayDelete}
+        keepMounted={true}
+        TransitionComponent={Transition}
+        onClose={handleCloseDelete}
+      >
+        <DialogContent>
+          <DialogContentText>
+            {t_i18n('Do you want to delete this external reference?')}
+            {isExternalReferenceAttachment && (
+              <Alert
+                severity="warning"
+                variant="outlined"
+                style={{ position: 'relative', marginTop: 20 }}
+              >
+                {t_i18n(
+                  'This external reference is linked to a file. If you delete it, the file will be deleted as well.',
+                )}
+              </Alert>
+            )}
+          </DialogContentText>
+          {isExternalReferenceAttachment && (
+            <Alert
+              severity="warning"
+              variant="outlined"
+              style={{ position: 'relative', marginTop: 20 }}
+            >
+              {t_i18n(
+                'This external reference is linked to a file. If you delete it, the file will be deleted as well.',
+              )}
+            </Alert>
+          )}
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleCloseDelete} disabled={deleting}>
+            {t_i18n('Cancel')}
+          </Button>
+          <Button color="secondary" onClick={submitDelete} disabled={deleting}>
+            {t_i18n('Delete')}
+          </Button>
+        </DialogActions>
+      </Dialog>
+    </React.Fragment>
+  );
+};
+
+export default ExternalReferencePopoverDeletion;
```

Comment:
Could you change the file name as it is not a popover component?

Reply:
@marieflorescontact - Fixed/Updated - ExternalReferenceDeletion.tsx now